thoth/storages/graph/dgraph.py

```diff
@@ -1643,7 +1643,8 @@ def _create_python_package_record(
             entity, _ = self.create_python_package_version_entity(
                 python_package_version.package_name,
                 python_package_version.package_version,
-                python_package_version.index_url
+                python_package_version.index_url,
+                not python_package_version.solver_error_unsolvable
```

Comment:
When we are creating `PythonPackageVersionEntity` if `solver_error_unsolvable` is True, then it means it is not present on index, so the attribute `is_present_on_index` for this entity would be `False`. Further I think we could change our query so only grab those `PythonPackageVersionEntity`s whose `is_present_on_index` is True and run `package-analyzer` jobs for them only.

Reply:
A package can be unsolvable due to other issues (e.g. Python interpreter version incompatibiliy). The `is_present_on_index` error is a subset of this issue. PTAL on this PR which will introduce a new flag in solver results which we can sync to the graph database [1].

[1] https://github.com/thoth-station/solver/pull/281